### PR TITLE
Ieee 146 only count healthy items towards quantity remaining

### PR DESCRIPTION
**Pull request by YuYing-Liang**
## Overview

- Resolves #IEEE-146
- Removed order items with part returned health of "Healthy" from checked out items in the annotation

## Unit Tests Created

- added a unit test to verify items returned with status other than Healthy or not counted to quantity remaining

## Steps to QA

- Create an order w/ order items on admin dashboard
- mark that order as picked up
- call the hardware api to see the quantity_remaining field change (decrease by however many you checked out) for a hardware item
- mark the order item has Healthy
- call hardware api again, quantity_remaining should've increased by 1

### CODE REVIEW

File: hackathon_site/hardware/tests.py [hunk 1]
```diff
@@ -121,28 +121,51 @@ def test_base_case_no_order_items(self):
     def test_some_items_none_returned(self):
         team = Team.objects.create()
         order = Order.objects.create(status="Picked Up", team=team)
```
Comment (Leo6Leo) [inline hackathon_site/hardware/tests.py:123]: the order model has been changed, there is one more field called "requests". And I will merge that pr asap!

[https://github.com/ieeeuoft/hackathon-template/pull/332](url)

File: hackathon_site/hardware/tests.py [hunk 2]
```diff
@@ -121,28 +121,51 @@ def test_base_case_no_order_items(self):
...
+            order=order, hardware=self.hardware,
+        )
         self.hardware.refresh_from_db()
         hardware_serializer = HardwareSerializer(self.hardware)
         self.assertEqual(hardware_serializer.data["quantity_remaining"], 2)
 
-    def test_some_items_returned(self):
+    def test_some_items_returned_healthy(self):
         team = Team.objects.create()
         order = Order.objects.create(status="Picked Up", team=team)
```
Comment (Leo6Leo) [inline hackathon_site/hardware/tests.py:136]: order obj here! New request field needed here

File: hackathon_site/hardware/tests.py [hunk 3]
```diff
@@ -121,28 +121,51 @@ def test_base_case_no_order_items(self):
...
+        OrderItem.objects.create(
+            order=order, hardware=self.hardware,
+        )
         self.hardware.refresh_from_db()
         hardware_serializer = HardwareSerializer(self.hardware)
         self.assertEqual(hardware_serializer.data["quantity_remaining"], 3)
 
+    def test_some_items_returned_not_healthy(self):
+        team = Team.objects.create()
+        order = Order.objects.create(status="Picked Up", team=team)
```
Comment (Leo6Leo) [inline hackathon_site/hardware/tests.py:149]: order obj here! New request field needed here

File: hackathon_site/hardware/tests.py [hunk 4]
```diff
@@ -121,28 +121,51 @@ def test_base_case_no_order_items(self):
...
+        OrderItem.objects.create(
+            order=order, hardware=self.hardware,
+        )
+        self.hardware.refresh_from_db()
+        hardware_serializer = HardwareSerializer(self.hardware)
+        self.assertEqual(hardware_serializer.data["quantity_remaining"], 2)
+
     def test_some_items_cancelled(self):
         team = Team.objects.create()
         order = Order.objects.create(status="Cancelled", team=team)
```
Comment (Leo6Leo) [inline hackathon_site/hardware/tests.py:162]: order obj here! New request field needed here

**Comment by Leo6Leo** [comment]
Other than that, I think all good!! 🎉

**Comment by YuYing-Liang** [comment]
@Leo6Leo since you are adding the new property for orders, you'll probably have to change all the unit tests to reflect that so when your changes are merged & I sync with develop, the unit tests in my PR will have the request field

**Comment by Leo6Leo** [comment]
> 

Oh that makes sense!

**Review by grahamhoyes — DISMISSED** [review]

**Review by grahamhoyes — APPROVED** [review]

**Pull request merged**